pkg/curl/schemes.go

```diff
@@ -356,17 +361,16 @@ func (s *SchemeWithRetries) Fetch(ctx context.Context, u *url.URL) (io.ReaderAt,
 
 // FetchWithoutCache implements FileScheme.FetchWithoutCache for retry wrapper.
 func (s *SchemeWithRetries) FetchWithoutCache(ctx context.Context, u *url.URL) (io.Reader, error) {
+	var r io.Reader
 	var err error
+
 	s.BackOff.Reset()
 	back := backoff.WithContext(s.BackOff, ctx)
-	for d := time.Duration(0); d != backoff.Stop; d = back.NextBackOff() {
-		if d > 0 {
-			time.Sleep(d)
-		}
-
-		var r io.Reader
-		// Note: err uses the scope outside the for loop.
-		r, err = s.Scheme.FetchWithoutCache(ctx, u)
+	for d := back.NextBackOff(); d != backoff.Stop; d = back.NextBackOff() {
```

Comment:
same question < instead of !=? Also, not your code, but these duplicated code blocks hint at a possibility that this could might be done differently? Not germane to this PR.